src/draft-service.ts

```diff
@@ -416,38 +477,65 @@ export async function deleteDraft(
       },
     )
   }
-  return getDraftsData()
-    .then((draftsData) => {
-      const draft = draftsData.drafts.find((draft) => draft.draftId === draftId)
-      if (!draft) {
-        console.log('Could not find existing draft to delete in drafts', {
-          draftId,
-          draftsData,
-        })
-        return
-      }
-      draftsData.drafts = draftsData.drafts.filter(
-        (draft) => draft.draftId !== draftId,
+  try {
+    await removeLocalDraftSubmission(formSubmissionDraftId)
+    const localDraftsStorage = await getLocalDrafts()
+    const formSubmissionDraft =
+      localDraftsStorage.syncedFormSubmissionDrafts.find(
+        ({ id }) => id === formSubmissionDraftId,
       )
-      return removeDraftData(draft.draftDataId)
-        .then(() =>
-          utilsService.localForage.setItem(`DRAFTS_${username}`, draftsData),
+    if (formSubmissionDraft) {
+      const { hasDeletedRemoteDraft } = await deleteDraftData(
+        formSubmissionDraftId,
+        abortSignal,
+      )
+      localDraftsStorage.syncedFormSubmissionDrafts =
+        localDraftsStorage.syncedFormSubmissionDrafts.filter(
+          ({ id }) => id !== formSubmissionDraft.id,
         )
-        .then(() => executeDraftsListeners(draftsData))
-        .then(() =>
-          syncDrafts({
-            throwError: false,
-            formsAppId,
-          }),
+      if (!hasDeletedRemoteDraft) {
+        localDraftsStorage.deletedFormSubmissionDrafts.push(formSubmissionDraft)
+      }
+    } else {
+      console.log('Could not find existing draft in synced drafts to delete', {
+        formSubmissionDraftId,
+        localDraftsStorage,
+      })
+      const draftSubmission = localDraftsStorage.unsyncedDraftSubmissions.find(
+        (draftSubmission) =>
+          draftSubmission.formSubmissionDraftId === formSubmissionDraftId,
+      )
+      if (!draftSubmission) {
+        return
+      }
```

Comment:
```suggestion
```

Reply:
Notice the `return`, want to stop and not finish function if draft does not exist.

Reply:
I don't see any negative side effects from continuing